sycl/test/sub_group/generic-shuffle.cpp

```diff
@@ -216,7 +213,7 @@ void check_struct(queue &Queue, Generator &Gen, size_t G = 256, size_t L = 64) {
 
 int main() {
   queue Queue;
-  if (!Queue.get_device().has_extension("cl_intel_subgroups")) {
+  if (Queue.get_device().is_host()) {
```

Comment:
Strictly speaking, this patch doesn't bring shuffles support to all non-host devices and "cl_intel_subgroups" extension is still required, but only for non-CUDA devices.

Reply:
You're right.  I wasn't sure what to do here, really -- the fact that "cl_intel_subgroups" is required is backend-specific.

@gmlueck: Do we have the necessary infrastructure implemented to query whether a particular extension from sycl/docs/extensions is supported for a given device?

Reply:
I guess you are thinking of adding a new "aspect" and then checking it like:

```
if (Queue.get_device().has(aspect::ext_intel_has_subgroups)) {
```

Correct?

@glyons-intel has added support for some aspects already, so it probably wouldn't be too hard to add another.

Reply:
Thanks, @gmlueck.  That's the sort of thing I was thinking of.  But as you noted elsewhere, we really have two options:
1) Define a new SYCL aspect describing devices that implement the sub-group extension
2) Commit to the path that all SYCL devices are intended to implement sub-groups

Option 1) addresses @AlexeySachkov's concern about whether the tests are accurately reflecting current implementation.  But I think I prefer 2) as the correct long-term direction.  Sub-groups aren't an optional feature in SYCL 2020 provisional, and we don't expect users to have to check any sort of extension before using them.

If we go with 1), we should probably define aspects for all of our device-specific extensions and update all the tests accordingly.  If we go with 2), we should phase out our use of `__spirv_SubgroupShuffleINTEL` in favor of the standard `__spirv_OpGroupNonUniformShuffle`.

Both of these seem fairly big jobs and outside of the scope of this PR.  @AlexeySachkov, are you okay to defer the resolution here until a future PR?

Reply:
While working on the group algorithms, I also stumbled across this check: https://github.com/intel/llvm/blob/sycl/sycl/test/group-algorithm/reduce.cpp#L67.  I'd be okay with implementing something similar here in the short-term, if that's preferred.

Reply:
> @AlexeySachkov, are you okay to defer the resolution here until a future PR?

Sure, no objections